core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -649,6 +651,13 @@ public XContentBuilder field(XContentBuilderString name, Map<String, Object> val
         value(value);
         return this;
     }
+    
+    public XContentBuilder field(String name, Path value) throws IOException {
```

Comment:
Even if it is a corner-case, the `field(String, Iterable)` method would still get called if you do something like:

```java
Path path = ...;
Iterable iterable = path;
builder.field("path", iterable); // this will call field(String, Iterable), not field(String, Path)
```

So I don't think we should try to add these specialized methods for `Path` and just make make sure that `Path` instances are escaped in all XContentBuilder methods that either take an `Iterable` or perform an `instanceof Iterable` test?